### ISSUE TITLE
VIP: Structs as arguments

### ISSUE THREAD

**Issue by fubuloubu** [ethereum/vyper#1019]
## Simple Summary
Accept Structs are args.
Depends on #300

## Abstract
It would be way easier to operate with different datatypes if we had better ways of interacting with more complex types like structs.

## Motivation
Solidity allows accepting structs as arguments via their experimental `ABIEncoderV2` pragma.
I believe v0.5 will allow this without an experimental pragma, and it will come out soon.
We should adopt this standard to make it easier to work with structs in different ways.

## Specification
```python
# MyStruct is a struct type as defined in #300
def accepts_struct(a: MyStruct):
    ...
```
(Optional) allow structs to be returned as outputs.
```python
def returns_struct() -> MyStruct:
    ...
```

## Backwards Compatibility
No modifications of existing syntax
ABI Encoder will need to be modified, but should be backwards compatible

## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/)

**Comment by jacqueswww** [comment]
Approved, dependency on #300.

**Comment by charles-cooper** [comment]
Remaining stuff after #1115 is merged

- Pass in structs to public functions (requires eth-abi updates to test)
- Return nested structs from functions (requires eth-abi updates to test)
- Pass nested structs as args
- Pass/return structs with dynamic data

**Comment by jacqueswww** [comment]
Meeting Note: When eth-tester relase gets cut, we can look at implementing the final steps listed by Charles.

**Comment by fubuloubu** [comment]
We should be able to finish the work suggested as remaining in this PR at this point.

**Comment by charles-cooper** [comment]
yeah we could punt to another issue

**Comment by fubuloubu** [comment]
Closing as the majority of this functionality was implemented in #1115

Note that some corner cases exist, whose progress is now being tracked in #955/#1200 (same resolution), #1170, and #1648
